### PR TITLE
Steven barrett

### PR DESCRIPTION
**Pull request by steven-barrett**
Completed the assignments and the stretch items

### CODE REVIEW

File: assignments/prototype-refactor.js [hunk 1]
```diff
@@ -7,3 +7,145 @@ Prototype Refactor
...
+  */
+  
+  class CharacterStats extends GameObject {
+      constructor(characterAttributes) {
+          super(characterAttributes);
+          this.healthPoints = characterAttributes.healthPoints;  
+        }
+  }  
+  CharacterStats.prototype.takeDamage = function () {
+    return `${mage.name} took damage.`;
```
Comment (adammathieson) [inline assignments/prototype-refactor.js:55]:
```suggestion
    return `${this.name} took damage.`;
```

File: assignments/prototype-refactor.js [hunk 2]
```diff
@@ -7,3 +7,145 @@ Prototype Refactor
...
+   class Humanoid extends CharacterStats {
+     constructor(humanAttributes) {
+        super(humanAttributes);
+        this.team = humanAttributes.team;
+        this.weapons = humanAttributes.weapons;
+        this.language = humanAttributes.language;   
+        }
+   }
+   Humanoid.prototype.greet = function () {
+    return `${archer.name} offers a greeting in ${archer.language}.`
```
Comment (adammathieson) [inline assignments/prototype-refactor.js:76]:
```suggestion
    return `${this.name} offers a greeting in ${archer.language}.`
```

Comment (adammathieson) [inline assignments/prototype-refactor.js:76]:
```suggestion
    return `${archer.name} offers a greeting in ${this.language}.`
```

File: assignments/lambda-classes.js [hunk 3]
```diff
@@ -1 +1,180 @@
...
+        console.log(`${this.name} has submitted a PR for ${subject}`);
+    }
+    sprintChallenge(subject) {
+        console.log(`${this.name} has begun sprint challenge on ${subject}`);
+    }
+    graduate() {
+        if (this.grade === 0) {
+            console.log(`${this.name} has completely failed at Lambda school!!!`)
+        }
+        else {
```
Comment (adammathieson) [inline assignments/lambda-classes.js:55]: Best practice syntax is -
graduate() {
        if (this.grade === 0) {
            console.log(`${this.name} has completely failed at Lambda school!!!`)
        } else if (this.grade > 70) {
                console.log(`${this.name} has graduated from Lambda school!!!`)
            } else {
                console.log(`${this.name} is not ready to graduate.`)
            }
        }

Yours works fine, readability issue

**Review by adammathieson — COMMENTED** [review]
The functions appear to work, however with the specific objects set to properties - the fn. will only make sense when called on those objects. The classes exercise does not have the same bug, so i'll assume it was another oversight. 
Classed looks great. Just one comment on syntax choice, but really great job, you know the code well

**Comment by steven-barrett** [comment]
I see what you mean about specific objects set to properties, I forgot to
change the code to it using “this” instead of specific objects.

**Pull request merged**